src/main/java/com/google/codeu/servlets/MessageServlet.java

```diff
@@ -79,14 +97,53 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     }
 
     String user = userService.getCurrentUser().getEmail();
-    String text = Jsoup.clean(request.getParameter("text"), Whitelist.relaxed());
+    String userText = Jsoup.clean(request.getParameter("text"), Whitelist.none());
+
+    String regexImage = "(https?://\\S+\\.(png|jpg|gif))";
+    String replacementImage = "<img src=\"$1\" />";
+    String regexVideo = "(https?://www.youtube.com/\\S+)";
+    String replacementVideo = "<iframe width=\"420\" height=\"345\" src=\"$1\">" + "</iframe>";
+
+    String textForImageAndVideo = userText;
+
+    // Validation of URL
+    Pattern patternImg = Pattern.compile(regexImage);
+    Pattern patternVid = Pattern.compile(regexVideo);
+    Matcher matcherImg = patternImg.matcher(userText);
+    Matcher matcherVid = patternVid.matcher(userText);
+
+    // Checks if the URL is valid and if it´s then it changes to insert the image
+    if (matcherImg.find()) {
+      try {
+        new URL(matcherImg.group()).toURI();
+        textForImageAndVideo = userText.replaceAll(regexImage, replacementImage);
+      } catch (URISyntaxException exception) {
+        // Not valid URL
+      } catch (MalformedURLException exception) {
+        // Not valid URL
+      }
+    }
```

Comment:
Hmm, wondering if we can use that UrlValidator function here?

Reply:
The only UrlValidator function I found was in apache library but because it was such a trouble to install it, I changed the approach to the Java URL creator so in case the URL isn´t valid Java will not be able to create the URL and throw the errors

Reply:
Oh I now see what are you talking about let me change it